### ISSUE TITLE
About the loss function and boundary conditions

### ISSUE THREAD

**Issue by RondoWong** [JinshuaiBai/LSWR_loss_function_PINN#6]
Hello Dr.Bai:
I'm a user of Torch instead of Tensorflow. I have some difficulties when I‘m trying to redo the pure bending beam case of your research work using Torch.  It's my pleasure if you can answer the following questions for me.

1.There are 2 displacement boundary conditions in your paper. The one for displacement in x direction is satisfied naturally through 'U = temp * x' in Dif_u. However I didn't see the imposition of V(0.5,0)=0. And as shown in the figure, I suppose  V(0,0)=0 is correct but not  V(0.5,0)=0.

2.Is it because of symmetry, the sheer stress is 0 and the normal stress is unknown at the left boundary？So the loss function is defined as l2.

3.I have noticed that you predict the displacement in y direction as 't = net2(xy_r)' corresponding to pinn(x)[10] in Loss_grad. But it seems that this item didn't make any contribution to the loss function. I wanna why you do that.

**Comment by JinshuaiBai** [comment]
For your questions:
1.   I do not explicitly impose V(0.5,0)=0 in the training.   I regard this as a rigid displacement and is achieved in the Vis function.   As you mentioned, in fact V(0,0)=0 is used but V(0.5,0)=0.   This is my mistake.   Thanks for pointing it out!
2.   Yes, you are right.   This is the pre-knowledge gained by mechanics analysis.
3.   Indeed, we do not include 't = net2(xy_r)' in the final loss function.   This was originally prepared for energy-based loss.

I have seen a lot of work regarding PINNs using PyTorch, it seems a trend.   It is good to have your own torch code.   Besides, writing code by yourself will help you have a better understanding of the methodology.

**Comment by RondoWong** [comment]
Thank you for your guidance! I will pay attention to your research work continually, which deeply enlightens me.
And it seems that my exploration doesn't go so smoothly. Anyway I will try it.
Moreover I want to redo the 3D case in your paper. Would you please send me the code? My email's address is wangyt0312@163.com. Thanks a lot!

**Comment by JinshuaiBai** [comment]
> Thank you for your guidance! I will pay attention to your research work continually, which deeply enlightens me. And it seems that my exploration doesn't go so smoothly. Anyway I will try it. Moreover I want to redo the 3D case in your paper. Would you please send me the code? My email's address is [wangyt0312@163.com](mailto:wangyt0312@163.com). Thanks a lot!

Sure, happy to know that my work helped. I will send you the code via email.

**Comment by 1965330785** [comment]
Hello Dr.Bai:
Today is Friday, March 7, 2025, in the Gregorian calendar, and it is the eighth day of the second lunar month in the year of the Yi Si Snake in the lunar calendar. The current time is 19:03.  
 
User's message:  
I am a Chinese undergraduate student studying PINN. I encountered difficulties in reproducing your 3D case, and the results have not been converging. Could you share the 3D case code with me? My email is: 1965330785@qq.com.  
Additionally, I tried to use this improved loss function for the one-dimensional tensile rod case in another paper of yours, but I found the error to be very large and couldn't find a solution. Do you have any suggestions for me?

**Comment by 1965330785** [comment]
Hello Dr.Bai:
I am a Chinese undergraduate student studying PINN. I encountered difficulties in reproducing your 3D case, and the results have not been converging. Could you share the 3D case code with me? My email is: 1965330785@qq.com.
Additionally, I tried to use this improved loss function for the one-dimensional tensile rod case in another paper of yours, but I found the error to be very large and couldn't find a solution. Do you have any suggestions for me?

**Comment by JinshuaiBai** [comment]
> Hello Dr.Bai: I am a Chinese undergraduate student studying PINN. I encountered difficulties in reproducing your 3D case, and the results have not been converging. Could you share the 3D case code with me? My email is: [1965330785@qq.com](mailto:1965330785@qq.com). Additionally, I tried to use this improved loss function for the one-dimensional tensile rod case in another paper of yours, but I found the error to be very large and couldn't find a solution. Do you have any suggestions for me?

The code has been sent. Please check your email address. 
For the question, I don't have enough information, I cannot provide you with suggestions.
